src/metabase/util/stats.clj

```diff
@@ -0,0 +1,315 @@
+  [x]
+  (cond
+    (= 0 x) "0"
+    (<= 1 x 5) "1-5"
+    (<= 6 x 10) "6-10"
+    (<= 11 x 25) "11-25"
+    (<= 26 x 50) "26-50"
+    (<= 51 x 100) "51-100"
+    (<= 101 x 250) "101-250"
+    (> x 250) "250+"))
+
+
+(defn- bin-large-number
+  "Return large bin number. Assumes positive inputs"
+  [x]
+  (cond
+    (= 0 x) "0"
+    (< x 1) "< 1"
+    (<= 1 x 10) "1-10"
+    (<= 11 x 50) "11-50"
+    (<= 51 x 250) "51-250"
+    (<= 251 x 1000) "251-1000"
+    (<= 1001 x 10000) "1001-10000"
+    (> x 10000) "10000+"))
+
+(defn- get-value-frequencies
+  "go through a bunch of maps and count the frequency a given key's values"
+  [many-maps k]
+  (frequencies (map k many-maps))
+  )
+
+(defn- histogram
+  "Bin some frequencies using a passed in binning function"
+  [binning-fn many-maps k]
+  (frequencies (map binning-fn (vals (get-value-frequencies many-maps k))))
+  )
+
+(def micro-histogram
+  "Return a histogram for micro numbers"
+  (partial histogram bin-micro-number))
+
+(def small-histogram
+  "Return a histogram for small numbers"
+  (partial histogram bin-small-number))
+
+(def medium-histogram
+  "Return a histogram for medium numbers"
+  (partial histogram bin-medium-number))
+
+(def large-histogram
+  "Return a histogram for large numbers"
+  (partial histogram bin-large-number))
+
+(defn- instance-start-date
+  "Pull up the first user account and use that date"
+  []
+  ((first (db/select 'User {:limit 1} {:order-by [[:date_joined :desc]]})) :date_joined))
+
+(defn- environment-type
+  "Figure out what we're running under"
+  []
+  (cond
+    (= (config/config-str :mb-client) :osx) :osx
+    (config/config-str :rds-hostname) :elastic-beanstalk
+    (config/config-str :database-url) :heroku ;; Putting this last as 'database-url' seems least specific
+    :default "unknown"))
+
+(defn- get-settings
+  "Figure out global info aboutt his instance"
+  []
+  {:version               (config/mb-version-info :tag)
+   :running_on            (environment-type)
+   :application_database  (config/config-str :mb-db-type)
+   :check_for_updates     (setting/get :check-for-updates)
+   :site_name             (not= (public-settings/site-name) "Metabase")
+   :report_timezone       (setting/get :report-timezone)
+   :friendly_names        (humanization/enable-advanced-humanization)
+   :email_configured      ((resolve 'metabase.email/email-configured?))
+   :slack_configured      ((resolve 'metabase.integrations.slack/slack-configured?))
+   :sso_configured        (boolean ((resolve 'metabase.api.session/google-auth-client-id)))
+   :instance_started      (instance-start-date)
+   :has_sample_data       (db/exists? 'Database, :is_sample true)})
+
+;; util function
+(def add-summaries
+  "add up some dictionaries"
+  (partial merge-with +))
+
+;; User metrics
+(defn- user-dims
+  "characterize a user record"
+  [user]
+  {:total 1
+   :active (if (user :is_active) 1 0) ;; HOW DO I GET THE LIST OF ALL USERS INCLUDING INACTIVES?
```

Comment:
actually `(db/select User)` or just `(User)` will return all users by default.

Also looks like you missed this one, remember to do `(:is_active user)` instead of vice versa to avoid `NullPointerExceptions`.

https://github.com/bbatsov/clojure-style-guide#keywords-as-fn-to-get-map-values